server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1396,10 +1396,13 @@ public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineExcepti
     }
 
     /**
-     * Returns the lastest Replication Checkpoint that shard received. Shards will return an EMPTY checkpoint before
-     * the engine is opened.
+     * Returns the latest ReplicationCheckpoint that shard received.
+     * @return EMPTY checkpoint before the engine is opened and null for non-segrep enabled indices
      */
     public ReplicationCheckpoint getLatestReplicationCheckpoint() {
```

Comment:
Should we change the signature to Optional<ReplicationCheckpoint> as well?

Reply:
>Should we change the signature to Optional as well?

Thanks @Bukhtawar for the review. Did you mean updating java doc to mention this method is applicable only for SegRep enabled indices ?